modules/pipeline/pipengine/actionexecutor/plugins/k8sflink/flink.go

```diff
@@ -66,20 +75,90 @@ func (k *K8sFlink) Status(ctx context.Context, task *spec.PipelineTask) (apistru
 		logrus.Errorf("get status err %v", err)
 
 		if strings.Contains(err.Error(), "not found") {
-			statusDesc.Status = apistructs.StatusNotFoundInCluster
+			statusDesc.Status = logic.TransferStatus(string(apistructs.StatusNotFoundInCluster))
 			return statusDesc, nil
 		}
-		statusDesc.Status = apistructs.StatusError
-		statusDesc.Reason = err.Error()
+		statusDesc.Status = logic.TransferStatus(string(apistructs.StatusError))
+		statusDesc.Desc = err.Error()
 		return statusDesc, err
 	}
 
-	statusDesc = composeStatusDesc(flinkCluster.Status)
+	status := composeStatusDesc(flinkCluster.Status)
+	statusDesc.Status = logic.TransferStatus(string(status.Status))
+	statusDesc.Desc = status.Reason
 
 	return statusDesc, nil
 }
 
-func (k *K8sFlink) Create(ctx context.Context, task *spec.PipelineTask) (interface{}, error) {
+func (k *K8sFlink) Exist(ctx context.Context, task *spec.PipelineTask) (created bool, started bool, err error) {
```

Comment:
You copied these related code three times. It's extremely wrong.
You should add these logics to a common pkg.

Reply:
adjusted, add these logics to logic.status file